### PR TITLE
[Bug][AMS][ARCTIC-733]: Catalog cache is not updated after edit catalog information

### LINKED ISSUE #733: [Bug][AMS]: Catalog cache is not updated after edit catalog information
### What happened?

image

after iceberg catalog properties ( set a new path to warehouse) ,  AMS can't  display table of catalog under new warehouse dir.

### Affects Versions

0.4.0

### What engines are you seeing the problem on?

AMS

### How to reproduce

1. create an catalog ( metastore = hadoop, table format = iceberg, set a dir of warehouse)
2. create some tables under the warehouse dir. 
3. edit the catalog information, and set the warehouse dir to a new location.
4. Tables page still shows tables under the old location.

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?
fix #733 and fix add optimize queue cache error

## Brief change log

1. remove catalog cache when edit catalog
2. getqueue id from system DB when load optimize queue config in the first time

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/controller/CatalogController.java [hunk 1]
```diff
@@ -357,6 +358,7 @@ public static void updateCatalog(Context ctx) {
     try {
       CatalogMeta catalogMeta = constructCatalogMeta(info, oldCatalogMeta);
       catalogMetadataService.updateCatalog(catalogMeta);
+      CatalogUtil.removeCatalogCache(catalogMeta.getCatalogName());
```
[the remark this answers is not in the document]
Comment (hameizi) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/controller/CatalogController.java:361]: this is the only catalog cache in ams-server

**Review by Aireed — APPROVED** [review]
LGTM
